compiler/src/org.graalvm.compiler.lir/src/org/graalvm/compiler/lir/gen/ArithmeticLIRGeneratorTool.java

```diff
@@ -44,6 +44,8 @@
  */
 public interface ArithmeticLIRGeneratorTool {
 
+    LIRGenerator getLIRGen();
```

Comment:
Can we remove this override?  It sort of defeats the purpose of the interface to expose this here.  I think replacing ArithmeticLIRGeneratorTool with ArithmeticLIRGenerator in Hasher and HashFunction resolve the problem without requiring this override.

Reply:
good point! it does work, I've removed the override